src/preamble.js

```diff
@@ -1086,3 +1082,68 @@ function getCompilerSetting(name) {
+  if (Module['preInit']) {
+    if (typeof Module['preInit'] == 'function') Module['preInit'] = [Module['preInit']];
+    while (Module['preInit'].length > 0) {
+      Module['preInit'].pop()();
+    }
+  }
+#if ASSERTIONS
+  consumedModuleProp('preInit');
+#endif
+#endif
+
+  {{{ makeModuleReceive('arguments_', 'arguments') }}}
+  {{{ makeModuleReceive('thisProgram') }}}
+
+
+#if ASSERTIONS
+  // Assertions on removed incoming Module JS APIs.
+  assert(typeof Module['memoryInitializerPrefixURL'] == 'undefined', 'Module.memoryInitializerPrefixURL option was removed, use Module.locateFile instead');
+  assert(typeof Module['pthreadMainPrefixURL'] == 'undefined', 'Module.pthreadMainPrefixURL option was removed, use Module.locateFile instead');
+  assert(typeof Module['cdInitializerPrefixURL'] == 'undefined', 'Module.cdInitializerPrefixURL option was removed, use Module.locateFile instead');
+  assert(typeof Module['filePackagePrefixURL'] == 'undefined', 'Module.filePackagePrefixURL option was removed, use Module.locateFile instead');
+  assert(typeof Module['read'] == 'undefined', 'Module.read option was removed');
+  assert(typeof Module['readAsync'] == 'undefined', 'Module.readAsync option was removed (modify readAsync in JS)');
+  assert(typeof Module['readBinary'] == 'undefined', 'Module.readBinary option was removed (modify readBinary in JS)');
+  assert(typeof Module['setWindowTitle'] == 'undefined', 'Module.setWindowTitle option was removed (modify emscripten_set_window_title in JS)');
+  assert(typeof Module['TOTAL_MEMORY'] == 'undefined', 'Module.TOTAL_MEMORY has been renamed Module.INITIAL_MEMORY');
+  assert(typeof Module['ENVIRONMENT'] == 'undefined', 'Module.ENVIRONMENT has been deprecated. To force the environment, use the ENVIRONMENT compile-time option (for example, -sENVIRONMENT=web or -sENVIRONMENT=node)');
+  assert(typeof Module['STACK_SIZE'] == 'undefined', 'STACK_SIZE can no longer be set at runtime.  Use -sSTACK_SIZE at link time')
+#if !IMPORTED_MEMORY
+  // If memory is defined in wasm, the user can't provide it, or set INITIAL_MEMORY
+  assert(typeof Module['wasmMemory'] == 'undefined', 'Use of `wasmMemory` detected.  Use -sIMPORTED_MEMORY to define wasmMemory externally');
+  assert(typeof Module['INITIAL_MEMORY'] == 'undefined', 'Detected runtime INITIAL_MEMORY setting.  Use -sIMPORTED_MEMORY to define wasmMemory dynamically');
+#endif
+  {{{ makeRemovedModuleAPIAssert('asm', 'wasmExports', false) }}}
+  {{{ makeRemovedModuleAPIAssert('readAsync') }}}
+  {{{ makeRemovedModuleAPIAssert('readBinary') }}}
+  {{{ makeRemovedModuleAPIAssert('setWindowTitle') }}}
+  {{{ makeRemovedFSAssert('IDBFS') }}}
+  {{{ makeRemovedFSAssert('PROXYFS') }}}
+  {{{ makeRemovedFSAssert('WORKERFS') }}}
+  {{{ makeRemovedFSAssert('FETCHFS') }}}
+  {{{ makeRemovedFSAssert('ICASEFS') }}}
+  {{{ makeRemovedFSAssert('JSFILEFS') }}}
+  {{{ makeRemovedFSAssert('OPFS') }}}
+
+#if !NODERAWFS
+  {{{ makeRemovedFSAssert('NODEFS') }}}
+#endif
```

Comment:
These all turn into `var IDBFS = 'IDBFS is no longer included ...';` and will now be in the processModuleArgs function. If the user tries to use the variable it will undefined which will be different than the old behavior. Do we want to leave these outside of the function?